libs/core/langchain_core/callbacks/manager.py

```diff
@@ -2618,3 +2620,10 @@ def foo(inputs):
         data,
         run_id=callback_manager.parent_run_id,
     )
+
+
+@functools.lru_cache(maxsize=1)
+def _executor() -> ThreadPoolExecutor:
```

Comment:
By the way consider adding an inline doc-string for other maintainers.

This change on its own is very simple, but there's a lot of non trivial stuff happening in the larger context, so it could be nice to explain the reason why some things exist even if they seem self-explantory right now